src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,376 @@
+        pal_low_frequency = 3550000000.0
+        pal_high_frequency = 3560000000.0
+        pal_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+        ppa_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+        ppa, pal = makePpaAndPalRecordsConsistent(ppa_record,
+                                                                [pal_record],
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        ppas.append(ppa)
+        pals.extend(pal)
+        # ESC senosrs
+        esc_sensors = []
+        esc_sensors.append(json.load(
+              open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json'))))
+        # SAS Test Harness 
+        sas_harness_config = {
+          'sasTestHarnessName': 'SAS-TH-2',
+          'url': 'https://localhost:9003/v1.2',
+          'serverCert': os.path.join('certs', "sas.cert"),
+          'serverKey': os.path.join('certs', "sas.key")
+        }	
+        config = {
+          'registrationRequests': devices,
+          'conditionalRegistrationData': conditionals,
+          'grantRequests': grants,
+          'ppas': ppas,
+          'palRecords': pals,
+          'escSensors' : esc_sensors,
+          'sasHarness' : sas_harness_config
+        }
+        writeConfig(filename, config)
+    
+    @configurable_testcase(generate_FAD_1_default_config)
+    def test_WINNF_FT_S_FAD_1(self, config_filename):
+        """ This test verifies that a SAS UUT can successfully respond to a full
+             activity dump request from a SAS Test Harness
+			
+		    SAS UUT approves the request and responds,
+		    with correct content and format for both dump message and files 
+        """
+        # load config file
+        config = loadConfig(config_filename)
+        # Very light checking of the config file.
+        self.assertEqual(len(config['registrationRequests']),
+                         len(config['grantRequests']))
+        
+        # inject FCC IDs and User IDs of CBSDs   
+        for device in config['registrationRequests']:
+          self._sas_admin.InjectFccId({
+              'fccId': device['fccId'],
+              'fccMaxEirp': 47
+          })
+          self._sas_admin.InjectUserId({'userId': device['userId']})
+          
+        # Pre-load conditional registration data for N3 CBSDs.
+        self._sas_admin.PreloadRegistrationData(
+            config['conditionalRegistrationData'])
+        
+        # Register N1 CBSDs.
+        request = {'registrationRequest': config['registrationRequests']}
+        responses = self._sas.Registration(request)['registrationResponse']
+        # Check registration responses and get cbsd Id
+        grants = config['grantRequests']
```

Comment:
This probably belongs a bit lower down (just above line 289).

Reply:
yes, I fixed that, thank you.